test/libsolidity/semanticTests/strings/empty_storage_string_1.sol

```diff
@@ -0,0 +1,12 @@
+contract C {
```

Comment:
Some more test ideas to make sure we're not missing any other cases:
- Encoding an empty string from calldata and from memory.
- Encoding an empty string by returning it from external function (rather than emitting it in an event). Or including it in an error. Or using `abi.encode()` explicitly.
- Encoding empty `bytes`. Or an empty string/bytes literal.

Reply:
I see you added most of these but there could you add these as well?
- Emitting `""` literal directly (without assigning it anywhere).
- Storage variable of type `bytes`.
- The test that uses abicoder v1 from the original issue that returns correct results (unless we have that covered somewhere already).

Reply:
Sure! Working on it now.

Reply:
> I see you added most of these but there could you add these as well?
> 
>     * Emitting `""` literal directly (without assigning it anywhere).
Created a test function which emits an event object with the "" literal.
> 
>     * Storage variable of type `bytes`.
Created a test function which returns an uninitialized bytes variable from storage.
> 
>     * The test that uses abicoder v1 from the original issue that returns correct results (unless we have that covered somewhere already).
I guess this test is already present, it would be the test that emits an event with an empty string. Should I add a test with a line with "pragma abicoder v1;" ?